src/systems/diff_drive/DiffDrive.cc

```diff
@@ -422,16 +434,27 @@ void DiffDrivePrivate::UpdateOdometry(const ignition::gazebo::UpdateInfo &_info,
   // Set the frame id.
   auto frame = msg.mutable_header()->add_data();
   frame->set_key("frame_id");
-  frame->add_value(this->model.Name(_ecm) + "/odom");
+  if (this->sdf_frame_id.empty()){
+    frame->add_value(this->model.Name(_ecm) + "/odom");
+  }else{
+    frame->add_value(this->sdf_frame_id);
+  }
 
   std::optional<std::string> linkName = this->canonicalLink.Name(_ecm);
-  if (linkName)
-  {
+  if (this->sdf_child_frame_id.empty()){
```

Comment:
nit: `{` should be on a new line in accordance to the [style guide](https://ignitionrobotics.org/docs/all/contributing#style-guides).